src/streamlink/plugins/vimeo.py

```diff
@@ -8,70 +8,175 @@
+            elif not self.session.api._auth_data or time_now - self.session.api._last_updated > self._expiry_time:
+                log.debug("Reloading Vimeo auth data")
+                self._update_auth()
+
+            self._expiry = self.get_expiry(self.session.api._auth_data)
+            self._path_parts[1] = self.session.api._auth_data
+            self._url = self._parsed_url._replace(path="/".join(self._path_parts)).geturl()
+            log.debug("Reloaded Vimeo HLS URL")
+        return self._url
+
+
+class VimeoAPI:
     _config_schema = validate.Schema(
         validate.parse_json(),
         {
             "request": {
                 "files": {
-                    validate.optional("dash"): {"cdns": {validate.text: {"url": validate.url()}}},
-                    validate.optional("hls"): {"cdns": {validate.text: {"url": validate.url()}}},
+                    validate.optional("dash"): {"cdns": {validate.text: {"url": validate.any(None, validate.url())}}},
+                    validate.optional("hls"): {"cdns": {validate.text: {"url": validate.any(None, validate.url())}}},
                     validate.optional("progressive"): validate.all(
-                        [{"url": validate.url(), "quality": validate.text}]
+                        [{"url": validate.url(), "quality": validate.text}],
                     ),
                 },
                 validate.optional("text_tracks"): validate.all(
-                    [{"url": validate.text, "lang": validate.text}]
+                    [{"url": validate.text, "lang": validate.text}],
                 ),
-            }
+            },
         },
     )
-    _player_schema = validate.Schema(
-        validate.transform(_config_re.search),
-        validate.any(None, validate.Schema(validate.get(1), _config_schema)),
-    )
 
+    def __init__(self, session, url):
+        self.session = session
+        self.url = url
+        self._auth_data = None
+        self._last_updated = None
+        self._updating = False
+
+    def get_player_data(self):
+        return self.session.http.get(self.url, schema=validate.Schema(
+            validate.transform(re.compile(r"var\s+config\s*=\s*({.+?})\s*;").search),
+            validate.any(None, validate.Schema(validate.get(1), VimeoAPI._config_schema)),
+        ))
+
+    def get_api_url(self):
+        return self.session.http.get(self.url, schema=validate.Schema(
+            validate.transform(re.compile(r'(?:"config_url"|\bdata-config-url)\s*[:=]\s*(".+?")').search),
+            validate.any(
+                None,
+                validate.Schema(
+                    validate.get(1),
+                    validate.parse_json(),
+                    validate.transform(html_unescape),
+                    validate.url(),
+                ),
+            ),
+        ))
+
+    def get_config_data(self, api_url):
+        return self.session.http.get(api_url, schema=VimeoAPI._config_schema)
```

Comment:
Don't statically reference the `_config_schema` class attribute, use `self._config_schema`. Same thing above.